addons/google_calendar/models/google_calendar.py

```diff
@@ -869,125 +853,124 @@ def update_events(self, cr, uid, lastSync=False, context=None):
                 datetime.strptime(current_user.google_calendar_token_validity.split('.')[0], DEFAULT_SERVER_DATETIME_FORMAT) < (datetime.now() + timedelta(minutes=1)):
-            self.do_refresh_token(cr, uid, context=context)
+            self.do_refresh_token()
             current_user.refresh()
         return current_user.google_calendar_token
 
-    def get_last_sync_date(self, cr, uid, context=None):
-        current_user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+    def get_last_sync_date(self):
+        current_user = self.env['res.users'].browse(self.env.uid)
         return current_user.google_calendar_last_sync_date and datetime.strptime(current_user.google_calendar_last_sync_date, DEFAULT_SERVER_DATETIME_FORMAT) + timedelta(minutes=0) or False
 
-    def do_refresh_token(self, cr, uid, context=None):
-        current_user = self.pool['res.users'].browse(cr, uid, uid, context=context)
-        gs_pool = self.pool['google.service']
+    def do_refresh_token(self):
+        current_user = self.env['res.users'].browse(self.env.uid)
+        GsPool = self.env['google.service']
 
-        all_token = gs_pool._refresh_google_token_json(cr, uid, current_user.google_calendar_rtoken, self.STR_SERVICE, context=context)
+        all_token = GsPool._refresh_google_token_json(current_user.google_calendar_rtoken, self.STR_SERVICE)
 
         vals = {}
         vals['google_%s_token_validity' % self.STR_SERVICE] = datetime.now() + timedelta(seconds=all_token.get('expires_in'))
         vals['google_%s_token' % self.STR_SERVICE] = all_token.get('access_token')
 
-        self.pool['res.users'].write(cr, SUPERUSER_ID, uid, vals, context=context)
+        self.env['res.users'].sudo().browse(self.env.uid).write(vals)
 
-    def need_authorize(self, cr, uid, context=None):
-        current_user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+    def need_authorize(self):
+        current_user = self.env['res.users'].browse(self.env.uid)
         return current_user.google_calendar_rtoken is False
 
     def get_calendar_scope(self, RO=False):
         readonly = RO and '.readonly' or ''
         return 'https://www.googleapis.com/auth/calendar%s' % (readonly)
 
-    def authorize_google_uri(self, cr, uid, from_url='http://www.openerp.com', context=None):
-        url = self.pool['google.service']._get_authorize_uri(cr, uid, from_url, self.STR_SERVICE, scope=self.get_calendar_scope(), context=context)
+    def authorize_google_uri(self, from_url='http://www.odoo.com'):
+        url = self.env['google.service']._get_authorize_uri(from_url, self.STR_SERVICE, scope=self.get_calendar_scope())
         return url
 
-    def can_authorize_google(self, cr, uid, context=None):
-        return self.pool['res.users'].has_group(cr, uid, 'base.group_erp_manager')
+    def can_authorize_google(self):
+        return self.env['res.users'].has_group('base.group_erp_manager')
```

Comment:
@rga-odoo even there is no need of record set, has_group will always check user belongs to particular group or not based on current uid, so model is enough to accomplish job